core/src/com/unciv/models/ruleset/tile/TileImprovement.kt

```diff
@@ -137,7 +138,61 @@ class TileImprovement : RulesetStatsObject() {
             textList += FormattedLine("Unique to [$uniqueTo]", link="Nation/$uniqueTo")
         }
 
-        if (terrainsCanBeBuiltOn.isNotEmpty()) {
+        //todo Why does this have to be so complicated? A unit's "Can build [Land] improvements on tiles"
+        //     creates the _justified_ expectation that an improvement it can build _will_ have
+        //     `matchesFilter("Land")==true` - but that's not the case.
```

Comment:
That's exactly the point - how do you determine that _without_ a running game, from the ruleset alone? I'd be _happy_ to hear it can be done simpler. But I doubt it. These parts have grown on a needed-to-fix-kludges base, not designed to be a consistent set of rules.

That whole monstrosity of code is done half tongue-in-check to demonstrate something is rotten to the core, but I'm not the visionary needed to redesign that. But if you want a list "which units can build a specific improvement" from ruleset alone, this is what would work. Or - construct a headless mock game world like some unit tests do, create all possible terrains on it, place all possible units on them and do a statistics aggregation on what the actual canBuild* routines say...

Reply:
The `matchesFilter` of TileImprovement does not support that "Land" keyword at all.